CMakeLists.txt

```diff
@@ -121,12 +122,25 @@ add_compile_definitions(
     QT_USE_QSTRINGBUILDER
 )
 
+if(BUILD_WASM)
+    set(BUILD_TESTING OFF)
+    set(Rust_CARGO_TARGET wasm32-unknown-emscripten)
+endif()
+
 # QMAKE environment variable is needed by qt-build-utils to ensure that Cargo
 # uses the same installation of Qt as CMake does.
 if(NOT USE_QT5)
     find_package(Qt6 COMPONENTS Core Gui Test)
 endif()
 if(NOT Qt6_FOUND)
+    if(BUILD_WASM)
+        message("
```

Comment:
can we instead just fail the build ? eg have a `message(FATAL_ERROR "...")` ?

Reply:
sure